index.js

```diff
@@ -31,17 +43,20 @@ class Chunks2JsonWebpackPlugin {
     }
     saveJson() {
         const projectRoot = process.cwd();
-        let pathStep = projectRoot;
-        this.options.outputDir.replace(projectRoot, '').split('/').forEach((folder) => {
-            pathStep = path.join(pathStep, folder);
-            try {
-                fs.mkdirSync(pathStep);
-            } catch (e) {
-                // we don't care if it already exists, just continue...
-            }
-        });
-        const file = path.join(process.cwd(), this.options.outputDir, this.options.filename);
-        const blob = JSON.stringify(this.result, undefined, 2);
+        // try to create outputDir folder if it is within project root
+        if (this.options.outputDir.startsWith(projectRoot) || !this.options.outputDir.startsWith('/')) {
+            let pathStep = projectRoot;
+            this.options.outputDir.replace(/^\.\//, '').replace(projectRoot, '').split('/').forEach((folder) => {
```

Comment:
I think is time this is split in multiple steps. As if you try to read this, is not clear right away what it does. 

Something like this is nicer (not sure about naming though :) ): 

```javascript
// remove relative prefix
const pathWithoutRelativePrefix = this.options.outputDir.replace(/^\.\//, '');
// remove path prefix (in case absolute path is passed to options) to current folder
const pathFromCurrentFolder = pathWithoutRelativePrefix.replace(projectRoot, '');
pathFromCurrentFolder.split('/').forEach((folder) => {

}
```